### DISCUSSION TITLE
ffmpeg tradeoffs and troubles with go2rtc vs. frigate

### DISCUSSION THREAD

**Discussion by natevw** [blakeblackshear/frigate#8606] [Ask A Question]
Sorry for another somewhat vague question, I think I might have to sooner or later spin up a separate Docker compose stack with a second copy of Frigate and a simpler config file just to fully debug each camera one at a time. But some of the general trouble I'm having is that for some reason the Frigate core detection and the "Cameras" thumbnails and Birdseye and all seem to work best when I pass raw `rtsp:` URLs to nearly all my cameras (there's *one* that just doesn't seem to work unless it's `ffmpeg:` from the get-go…).

But the tradeoff is that because most of these cameras are H.265 (and/or a variety of different audios) this means that I can only reliably use the "mjpeg" option to watch them live. On some computers I can also use the "mse" option but it's kind of hit or miss depending on if it's a Chromebook or Firefox/Linux or whatever I happen to pull it up on.

Here's where it starts to get hard to chase down! If I leave a camera in the go2rtc with the raw feed and try to add ffmpeg:

```
mycam:
- rtsp://some-cam.lan:554/stream0
- ffmpeg:mycam#video=h264#audio=opus#hardware
```

it tends to start ~crashing (breaking somehow, anyway…) apparently at the go2rtc level (all the JSON info just becomes `null` as soon as anything tries to connect). Like go2rtc really hates to connect ffmpeg to a *named* feed [unless that feed was `ffmpeg:` to start with]?

But if I switch the camera to ffmpeg right away:

```
mycam:
- ffmpeg:rtsp://some-cam.lan:554/stream0
- ffmpeg:mycam#video=h264#audio=opus#hardware
```

that seems to go better on the go2rtc side, but makes Frigate really unreliable. The ffmpeg streams seem to take forever to load. Sometimes that's an exaggerated "forever", and really I just have to wait a couple minutes (but still super annoying when restarting all the time to edit config!). But other times it seems to be an *actual* "forever" in that Frigate just stays disconnected indefinitely — until I click the camera feed and view it through its seems maybe *any* of mjpeg/webrc/mse and then **finally** the main parts of Frigate wake up and **finally** stop showing that camera with a broken "No frames have been received". (So essentially if the server reboots for any reason the camera will be offline for hours if not days, unless and until I manually intervene.)

Again, I realize there's a ton of variables here, but just wondering if any of this sounds familiar or expected? I'd like for both of these to be true simultaneously:

1. can negotiate suitable re-encodings for the mse/webrtc options so that I can watch any camera live from any halfway-modern browser
2. but also that Frigate reliably always gets frames from all the cameras after a restart, ideally very soon after startup but some reasonable definition of "eventually" would probably be acceptable

What I don't really get is — Frigate itself uses ffmpeg to read and record the feeds, right? So why would it make such a difference, both good and bad, where ffmpeg first gets configured into the mix?

**Comment by NickM-27** [comment]
What version of frigate are you running, if it's 0.12 then you might just want to update go2rtc.

**Reply by natevw** [reply]
I was runnning 0.12 but problems persisted even after I manually installed go2rtc v1.8.2:

```
- name: Custom go2rtc with more recent updates
    # see https://docs.frigate.video/configuration/advanced#custom-go2rtc-version
    ansible.builtin.get_url:
        url: "{{ go2rtc_dl_url }}"
        dest: /srv/frigate/config/go2rtc
        mode: 0766
```

Now with Frigate 0.13.0-beta5 I guess it's probably using that same "override" although iiuc it's the same version that's now bundled with the image either way? The problem with the cameras just **never** loading (i.e. remaining on "No frames have been received") actually seems to have been *introduced* by upgrading one or the other… but I've been doing tons of tweaking and adding new cameras and such so it's been fairly unscientific so far. Sounds like I might have to just start eliminating variables until I can ask some more specific questions about each of the specific things I've been bumping up against.

**Reply by Cold-Lemonade** [reply]
@natevw Are you still experiencing this problem? I am experiencing something very similar on 4 Anpviz cameras. Frigate stops recording their video feeds periodically because the ffmpeg stops.
